imagepipeline/src/main/java/com/facebook/imagepipeline/producers/DecodeProducer.java

```diff
@@ -439,6 +443,10 @@ protected synchronized boolean updateDecodeJob(EncodedImage encodedImage, boolea
       return ret;
     }
 
+    private boolean isNotPartialData(EncodedImage encodedImage) {
+      return encodedImage.getEncodedCacheKey() != null;
```

Comment:
Can't use the ```isLast``` here, as this encoded image could came from disk cache produced by media-variations fallback producer, in that case, value of ```isLast``` could be true if the cached image is large enough than the request specified, or false if a full network fetch is needed.
Most importantly, only when ```isLast == false``` will this method be executed, because if it is a last result of a request, the problem I described in the PR should never occur.